### PR TITLE
Merge Today Widget feature branch into develop

### PR DESCRIPTION
**Pull request by Gio2018**
It's a very big PR, containing all the previously merged PRs of the [project](https://github.com/wordpress-mobile/WordPress-iOS/projects/18)

Fixes #NA

To test:

- build/run and make sure you can install the today widget and it works as expected.

PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/Classes/Stores/StatsInsightsStore.swift [hunk 1]
```diff
@@ -981,3 +985,84 @@ private extension InsightStoreState {
...
+            return
+        }
+
+        var homeWidgetTodayCache = HomeWidgetTodayData.read() ?? initializeHomeWidgetTodayData()
+
+        guard let oldData = homeWidgetTodayCache[siteID.intValue] else {
+            DDLogError("HomeWidgetToday: Failed to find a matching site")
+            return
+        }
+        // TODO - TODAYWIDGET: it might be better to move the blog updates in SiteStatsInformation
```
Comment (frosty) [inline WordPress/Classes/Stores/StatsInsightsStore.swift]: Is this TODO still valid?

File: WordPress/Classes/Stores/StatsInsightsStore.swift [hunk 2]
```diff
@@ -981,3 +985,84 @@ private extension InsightStoreState {
...
+                                                                    url: blog.url ?? oldData.url,
+                                                                    timeZone: blogService.timeZone(for: blog),
+                                                                    date: Date(),
+                                                                    stats: stats)
+
+        HomeWidgetTodayData.write(items: homeWidgetTodayCache)
+        WidgetCenter.shared.reloadTimelines(ofKind: WPHomeWidgetTodayKind)
+    }
+
+    // TODO - TODAYWIDGET: this method will likely be exposed in the future to handle the creation of the file at login
```
Comment (frosty) [inline WordPress/Classes/Stores/StatsInsightsStore.swift]: Is this TODO still valid?

File: WordPress/WordPressIntents/SitesDataProvider.swift [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+            return firstTitle < secondTitle
+
+        })
+    }
+
+    // MARK: - Default Site
+
+    private var defaultSiteID: Int? {
+        // TODO - TODAYWIDGET: taking the default site id from user defaults for now.
```
Comment (frosty) [inline WordPress/WordPressIntents/SitesDataProvider.swift]: Is this TODO still valid?

**Review by frosty — APPROVED** [review]
I've not reviewed all the code, as this has all previously been reviewed before. I did however pick out a few TODOs I noticed, in case they can be removed now.

I ran this on iOS 14.3 and everything worked as expected. I also checked iOS 13.7, and the old widgets are still working fine.

We discussed removing the Widgets button from Stats – will we look at removing or changing its behaviour in iOS 14 in another PR?

**Comment by Gio2018** [comment]
Thanks for catching those TODOs @frosty , they need to be removed, and will do in this PR. As for hiding the "Widget" button for iOS 14, I'll do a separate PR.

**Review by frosty — APPROVED** [review]
Great, thanks for removing those!

**Pull request merged**